src/python/pants/backend/java/compile/javac_test.py

```diff
@@ -77,6 +80,17 @@ def rule_runner() -> RuleRunner:
 )
 
 
+def expect_single_expanded_coarsened_target(
+    rule_runner: RuleRunner, address: Address
+) -> CoarsenedTarget:
+    expanded_target = rule_runner.request(Targets, [Addresses([address])]).expect_single()
+    coarsened_targets = rule_runner.request(
+        CoarsenedTargets, [Addresses([expanded_target.address])]
+    )
+    assert len(coarsened_targets) == 1
+    return coarsened_targets[0]
```

Comment:
This seems to test the `CoarsenedTarget` API rather than `javac`... can probably remove it?

Reply:
This isn't a test, it's a helper for a commonly reused pattern in the actual tests.  I _have_ to do this expansion every time because of the interface of `compile_java_source`.  I feel like the assertion is valuable even just for readability, in the same spirit as `Targets.expect_single()`.